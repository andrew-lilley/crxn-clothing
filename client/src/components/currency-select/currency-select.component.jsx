import React from 'react';
import Select from 'react-select';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectActiveCurrency } from '../../redux/currency/currency.selectors';
import { setCurrency, setCurrencyLangCode } from '../../redux/currency/currency.actions';
import { toggleCartHidden } from '../../redux/cart/cart.actions';
import { rebuildCart } from '../../redux/cart/cart.actions';
import { CurrencySelectorContainer } from './currency-select.styles';

export const CurrencySelector = ({ currency, dispatch }) => {

  const options = [
    { 
      value: 'default', 
      label: '$' 
    },
    { 
      value: 'en-gb', 
      label: '£' 
    },
    { 
      value: 'de', 
      label: '€' 
    }
  ];

  return (
    <CurrencySelectorContainer 
      onClick={() => {
        dispatch(toggleCartHidden(true));
      }}
    >
      <Select 
        options={options} 
        value={{ label: currency }}
        placeholder=''
        onChange={(e) => {
          dispatch(rebuildCart(e.value));
          dispatch(setCurrency(e.label));
          dispatch(setCurrencyLangCode(e.value));
        }}
      />
    </CurrencySelectorContainer>
  )
}

const mapStateToProps = createStructuredSelector({
  currency: selectActiveCurrency
});

export default connect(mapStateToProps)(CurrencySelector);