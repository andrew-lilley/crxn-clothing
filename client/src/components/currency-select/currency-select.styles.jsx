import styled from 'styled-components';

export const CurrencySelectorContainer = styled.div`
  width: 70px;
`;

// This is required for testing.
CurrencySelectorContainer.displayName = 'CurrencySelectorContainer';