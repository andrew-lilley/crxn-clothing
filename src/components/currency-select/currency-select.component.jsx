import React from 'react';
import Select from 'react-select';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectActiveCurrencyLangCode, selectActiveCurrency } from '../../redux/currency/currency.selectors';
import { setCurrency, setCurrencyLangCode } from '../../redux/currency/currency.actions';
import './currency-select.styles.scss';

const CurrencySelector = ({ currency, dispatch }) => {

  const options = [
    { value: 'default', label: '$' },
    { value: 'en-gb', label: '£' },
    { value: 'de', label: '€' },
  ];

  return (
    <div className='currency-selector'>
      <Select 
        options={options} 
        value={{ label: currency }}
        placeholder=''
        onChange={(e) => {
          dispatch(setCurrency(e.label));
          dispatch(setCurrencyLangCode(e.value));
        }}
      />
    </div>
  )
}

const mapStateToProps = createStructuredSelector({
  langCode: selectActiveCurrencyLangCode,
  currency: selectActiveCurrency
});

export default connect(mapStateToProps)(CurrencySelector);