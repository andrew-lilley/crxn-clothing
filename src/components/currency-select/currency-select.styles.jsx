import styled from 'styled-components';

export const CurrencySelectorContainer = styled.div`
width: 70px;
`;