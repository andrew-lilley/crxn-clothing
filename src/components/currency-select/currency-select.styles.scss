.currency-selector {
  width: 70px;
}
