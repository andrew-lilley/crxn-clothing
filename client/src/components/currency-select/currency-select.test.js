import React from 'react';
import { shallow } from 'enzyme';
import { CurrencySelector } from './currency-select.component';

describe('ErrorBoundary component', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      currency: 'default',
      dispatch: jest.fn()
    }
    wrapper = shallow(<CurrencySelector { ...mockProps } />);
  });

  test('should render CurrencySelector component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});