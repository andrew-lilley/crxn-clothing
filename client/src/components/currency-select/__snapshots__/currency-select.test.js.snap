// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component should render CurrencySelector component 1`] = `
<CurrencySelectorContainer
  onClick={[Function]}
>
  <StateManager
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={null}
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "$",
          "value": "default",
        },
        Object {
          "label": "£",
          "value": "en-gb",
        },
        Object {
          "label": "€",
          "value": "de",
        },
      ]
    }
    placeholder=""
    value={
      Object {
        "label": "default",
      }
    }
  />
</CurrencySelectorContainer>
`;
